source/server/listener_manager_impl.cc

```diff
@@ -484,15 +574,23 @@ ListenerImpl::findFilterChain(const Network::ConnectionSocket& socket) const {
   return nullptr;
 }
 
+namespace {
+
+// Return a fake address for use when either the source or destination is UDS.
+Network::Address::InstanceConstSharedPtr fakeAddress() {
+  static const Network::Address::InstanceConstSharedPtr address =
```

Comment:
Technically shared pointers are non-POD..

Reply:
There are many cases of this already in the code base, and there is no way this is an actual issue, but sure, will fix.